worlds/witness/hints.py

```diff
@@ -338,22 +340,85 @@ def get_priority_hint_locations(world: "WitnessWorld") -> List[str]:
     return priority
 
 
+def try_getting_location_group_other_world(world: "WitnessWorld", hint_loc: Location) -> str:
+    possible_location_groups = {
+        k: v for k, v in world.multiworld.worlds[hint_loc.player].location_name_groups.items()
+        if hint_loc.name in v
+    }
+
+    locations_in_that_world = {
+        location.name for location in world.multiworld.get_locations(hint_loc.player) if location.address is not None
+    }
+
+    valid_location_groups = dict()
+
+    for group, locations in possible_location_groups.items():
+        present_locations = sum(location in locations_in_that_world for location in locations)
+        if present_locations > 1:
+            valid_location_groups[group] = present_locations
+
+    if len(valid_location_groups) > 1 and valid_location_groups["Everywhere"] > 100:
+        del valid_location_groups["Everywhere"]
+
+    parent_region = hint_loc.parent_region
+    if parent_region.name not in possible_location_groups:
+        parent_region_location_amount = sum(
+            location.address is not None for location in parent_region.locations
+        )
+        if len(locations_in_that_world) / 2 > parent_region_location_amount > 2 and parent_region.name != "Menu":
```

Comment:
```suggestion
        if parent_region.name != "Menu" and 2 < parent_region_location_amount < (len(locations_in_that_world) / 2):
```
This is probably quite subjective, but I'd find this much more readable in reverse.